vault/expiration.go

```diff
@@ -936,16 +950,27 @@ func (m *ExpirationManager) RegisterAuth(source string, auth *logical.Auth) erro
 
 // FetchLeaseTimesByToken is a helper function to use token values to compute
 // the leaseID, rather than pushing that logic back into the token store.
-func (m *ExpirationManager) FetchLeaseTimesByToken(source, token string) (*leaseEntry, error) {
+func (m *ExpirationManager) FetchLeaseTimesByToken(source, tokenID string) (*leaseEntry, error) {
 	defer metrics.MeasureSince([]string{"expire", "fetch-lease-times-by-token"}, time.Now())
 
-	// Compute the Lease ID
-	saltedID, err := m.tokenStore.SaltID(m.quitContext, token)
+	idHMAC, err := m.tokenStore.hmac(m.quitContext, tokenID)
+	if err != nil {
+		return nil, err
+	}
+	le, err := m.FetchLeaseTimes(path.Join(source, idHMAC))
+	if err != nil {
+		return nil, err
+	}
+	if le != nil {
+		return le, nil
+	}
+
+	// This is only here for backwards compatibility
```

Comment:
This and the other one of these above almost suggest that maybe there should be a util method or something that handles this. Then there'd be only one place where the logic needs to be updated were there ever a version 3. I'm not totally sold on that myself, though, so if it makes the code odd or is a PIA, just ignore the idea. :-)

Reply:
Each place where I am using this is different. I couldn't see a common pattern that I could pull out, unfortunately.